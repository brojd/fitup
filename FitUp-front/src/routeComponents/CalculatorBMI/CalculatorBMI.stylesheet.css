.calcWrapper {
  background: black;
  height: 700px;
  overflowX: hidden;
  position: relative;
  margin-bottom: 200px;
  margin-top: 100px;
  transform: skew(-12deg);
  right: -100px;
  border-radius: 4px; }

.backImg {
  position: absolute;
  right: -300px;
  bottom: 0px;
  background: black;
  transform: skew(12deg); }

.calculator {
  transform: skew(12deg);
  backface-visibility: hidden;
  height: 700px; }

.mainHeading {
  font-weight: 500;
  letter-spacing: 1px;
  font-size: 32px;
  font-family: Papyrus, fantasy !important;
  color: white;
  line-height: 50px;
  transform: skew(12deg);
  backface-visibility: hidden; }

.mainHeadingWrapper {
  width: 300px;
  height: 50px;
  background: #960000;
  transform: skew(-12deg);
  top: -49px;
  position: relative;
  left: 79px;
  border-radius: 4px; }

.inputHeading {
  color: #FFFFFF;
  font-size: 28px;
  font-family: Papyrus, fantasy !important;
  margin-top: 30px; }

.input {
  height: 40px;
  background: #efefef;
  font-size: 24px;
  text-align: center;
  color: #383838;
  font-family: Papyrus, fantasy !important; }

.input::placeholder {
  color: rgba(130, 130, 130, 0.44);
  text-align: center; }

.input:focus::placeholder {
  opacity: 0; }

.alert {
  font-size: 18px;
  font-weight: 600;
  color: #bd3434;
  text-shadow: 0 0 2px rgba(158, 0, 0, 0.59); }

.calcButton {
  background: linear-gradient(to top right, rgba(169, 17, 17, 0.98), #630000);
  color: white;
  font-family: Papyrus, fantasy !important;
  font-size: 25px;
  height: 50px;
  width: 140px;
  border-radius: 3px;
  border: 1px solid #5f5f5f;
  margin-top: 30px; }

.calcButton:hover {
  cursor: pointer; }

.resultSection {
  background: linear-gradient(to top right, rgba(175, 0, 0, 0.74), #3e0000);
  padding: 10px;
  border-radius: 10px;
  box-shadow: 0 0 10px #ffffff;
  border: 1px solid #a70000;
  margin-top: 60px; }

.resultSection__line {
  color: #e6e6e6;
  font-size: 24px;
  font-family: Papyrus, fantasy !important;
  margin-top: 20px; }

.resultSection__line:last-child {
  margin-bottom: 20px; }

.resultSection__result {
  font-family: Papyrus, fantasy !important;
  font-size: 28px;
  line-height: 32px;
  letter-spacing: 1px;
  color: rgba(255, 242, 193, 0.95);
  padding: 10px; }

/*# sourceMappingURL=CalculatorBMI.stylesheet.css.map */
