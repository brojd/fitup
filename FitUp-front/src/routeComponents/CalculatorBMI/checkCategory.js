export default function checkCategory(resultBMI) {
  switch (true) {
    case (resultBMI < 15):
      return 'Very severely underweight';
    case (resultBMI < 16):
      return 'Severely underweight';
    case (resultBMI < 18.5):
      return 'Underweight';
    case (resultBMI < 25):
      return 'Normal (healthy weight)';
    case (resultBMI < 30):
      return 'Overweight';
    case (resultBMI < 35):
      return 'Obese Class I (Moderately obese)';
    case (resultBMI < 40):
      return 'Obese Class II (Severely obese)';
    default:
      return 'Obese Class III (Very severely obese)';
  }
}
