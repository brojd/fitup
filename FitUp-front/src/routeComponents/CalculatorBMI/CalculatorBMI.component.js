import React, { Component } from 'react';
import styles from './CalculatorBMI.stylesheet.css';
import classNames from 'classnames';
import checkCategory from './checkCategory';
import Image from '../../components/Image/Image.component';
import bmi_img from './bmi_img.png';

class CalculatorBMI extends Component {
  constructor() {
    super();
    this._handleCalculate = this._handleCalculate.bind(this);
    this._handleWeightChange = this._handleWeightChange.bind(this);
    this._handleHeightChange = this._handleHeightChange.bind(this);
    this.state = {
      resultVisible: false,
      weightValid: true,
      heightValid: true
    };
  }
  _handleWeightChange(e) {
    if (e.target.value < 1) {
      e.target.value = null;
      this.setState({
        weight: e.target.value,
        weightValid: false
      });
    } else {
      this.setState({
        weight: e.target.value,
        weightValid: true
      });
    }
  }
  _handleHeightChange(e) {
    if (e.target.value < 1) {
      e.target.value = null;
      this.setState({
        height: e.target.value,
        heightValid: false
      });
    } else {
      this.setState({
        height: e.target.value,
        heightValid: true
      });
    }
  }
  _handleCalculate(e) {
    e.preventDefault();
    let result = (this.state.weight / (Math.pow(this.state.height / 100, 2))).toFixed(2);
    this.setState({
      result: result,
      resultVisible: true
    });
  }
  render() {
    let weightAlert;
    let heightAlert;
    let resultSection;
    if (this.state.weightValid) {
      weightAlert = null;
    } else {
      weightAlert = <div className={classNames(styles.alert, 'uk-text-center uk-width-8-10 uk-align-center')}>
                      Enter number greater than 0
                    </div>;
    }
    if (this.state.heightValid) {
      heightAlert = null;
    } else {
      heightAlert = <div className={classNames(styles.alert, 'uk-text-center uk-width-8-10 uk-align-center')}>
                      Enter number greater than 0
                    </div>;
    }
    if (!this.state.resultVisible) {
      resultSection = null;
    } else {
      resultSection = <section className={classNames(styles.resultSection,
                                                     'uk-align-center uk-width-6-10 uk-text-center zoomBounceIn')}>
                        <div className={styles.resultSection__line}>
                          Result: <span className={styles.resultSection__result}>{this.state.result}</span>
                        </div>
                        <div className={styles.resultSection__line}>
                          Category: <span className={styles.resultSection__result}>{checkCategory(this.state.result)}</span>
                        </div>
                      </section>;
    }
    return (
      <div>
        <div className={classNames(styles.calcWrapper, 'uk-float-right uk-width-8-10 slideInRightSkew')}>
          <Image src={bmi_img} className={styles.backImg} width={1050}/>
          <section className={classNames(styles.calculator, 'uk-width-6-10')}>
            <div className={classNames(styles.mainHeadingWrapper)}>
              <h2 className={classNames(styles.mainHeading, 'uk-text-center')}>BMI CALCULATOR</h2>
            </div>
            <form>
              <div>
                <h3 className={classNames(styles.inputHeading, 'uk-text-center')}>Enter your weight (kg)</h3>
                <input type='number'
                       onChange={this._handleWeightChange}
                       placeholder='your weight...'
                       className={classNames(styles.input, 'uk-width-4-10 uk-align-center')}/>
                {weightAlert}
              </div>
              <div>
                <h3 className={classNames(styles.inputHeading, 'uk-text-center')}>Enter your height (cm)</h3>
                <input type='number'
                       onChange={this._handleHeightChange}
                       placeholder='your height...'
                       className={classNames(styles.input, 'uk-width-4-10 uk-align-center')} />
                {heightAlert}
              </div>
              <button onClick={this._handleCalculate}
                      className={classNames(styles.calcButton, 'uk-align-center')}>Calculate</button>
            </form>
            {resultSection}
          </section>
        </div>
      </div>
    );
  }
}

export default CalculatorBMI;
