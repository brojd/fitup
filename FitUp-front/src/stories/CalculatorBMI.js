import React from 'react';
import { storiesOf} from '@kadira/storybook';
import CalculatorBMI from '../routeComponents/CalculatorBMI/CalculatorBMI.component';

storiesOf('CalculatorBMI', module)
  .add('BMI calculator', () => (
    <CalculatorBMI />
  ))
