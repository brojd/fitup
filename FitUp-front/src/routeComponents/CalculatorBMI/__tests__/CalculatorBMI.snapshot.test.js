import React from 'react';
import CalculatorBMI from '../CalculatorBMI.component';
import renderer from 'react-test-renderer';

test('CalculatorBMI renders correctly', () => {
  
  const component = renderer.create(
    <CalculatorBMI />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
});
