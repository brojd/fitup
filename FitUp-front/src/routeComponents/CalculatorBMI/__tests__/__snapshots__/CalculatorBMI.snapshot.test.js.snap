exports[`test CalculatorBMI renders correctly 1`] = `
<div>
  <div
    className="calcWrapper uk-float-right uk-width-8-10 slideInRightSkew">
    <img
      alt={undefined}
      className="backImg"
      height={150}
      onClick={undefined}
      src="test-file-stub"
      width={1050} />
    <section
      className="calculator uk-width-6-10">
      <div
        className="mainHeadingWrapper">
        <h2
          className="mainHeading uk-text-center">
          BMI CALCULATOR
        </h2>
      </div>
      <form>
        <div>
          <h3
            className="inputHeading uk-text-center">
            Enter your weight (kg)
          </h3>
          <input
            className="input uk-width-4-10 uk-align-center"
            onChange={[Function]}
            placeholder="your weight..."
            type="number" />
        </div>
        <div>
          <h3
            className="inputHeading uk-text-center">
            Enter your height (cm)
          </h3>
          <input
            className="input uk-width-4-10 uk-align-center"
            onChange={[Function]}
            placeholder="your height..."
            type="number" />
        </div>
        <button
          className="calcButton uk-align-center"
          onClick={[Function]}>
          Calculate
        </button>
      </form>
    </section>
  </div>
</div>
`;
